src/averageAges.js

```diff
@@ -15,11 +15,25 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  let diedInCentury = people;
+
+  if (century !== undefined) {
+    diedInCentury = diedInCentury.filter(
+      person => Math.ceil(person['died'] / 100) === century
+    );
+  }
+
+  const menDiedInCentury = diedInCentury.filter(
+    person => person['sex'] === 'm'
+  );
+
+  const totalMenAge = menDiedInCentury.reduce(
+    (ageSum, person) => ageSum + person['died'] - person['born'], 0
```

Comment:
use destructuring.
redundant [] bracket syntax
```suggestion
    (ageSum, ({died, born })) => ageSum + died - born, 0
```